shared/domains/chelonia/internals.js

```diff
@@ -1631,9 +1688,13 @@ export default (sbp('sbp/selectors/register', {
         // we revert any changes to the contract state that occurred, ignoring this mutation
         console.warn(`[chelonia] Error processing ${message.description()}: ${message.serialize()}. Any side effects will be skipped!`)
         processingErrored = e?.name !== 'ChelErrorWarning'
-        this.config.hooks.processError?.(e, message, getMsgMeta(message, contractID, state))
-        // special error that prevents the head from being updated, effectively killing the contract
-        if (e.name === 'ChelErrorUnrecoverable') throw e
+        try {
+          if (!this.config.hooks.processError) throw e
+          this.config.hooks.processError(e, message, getMsgMeta(message, contractID, state))
+        } catch (e) {
+          // special error that prevents the head from being updated, effectively killing the contract
+          if (e.name === 'ChelErrorUnrecoverable') throw e
+        }
```

Comment:
This change, and the related `handleEventError` change below, has significantly complicated the code and is now also forcing developers to decide in their hooks how Chelonia's internals should behave. This is a clearly less simple, and seems to add unnecessary complexity to the codebase. Could these changes be reverted?

Reply:
I think they could since it seems like it's not needed after all.